src/app/services/EloCalculationService.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from '@angular/core';
+import { Player } from '../jscaip/Player';
+import { Utils } from '../utils/utils';
+import { EloInfo } from '../domain/EloInfo';
+
+export interface EloInfoPair {
+    playerZero: EloInfo,
+    playerOne: EloInfo,
+}
+
+export interface EloDifferences {
+    pointChangeForZero: number,
+    pointChangeForOne: number,
+}
+
+export interface EloEntry {
+    eloInfoPair: EloInfoPair,
+    winner: 'ZERO' | 'ONE' | 'DRAW',
+}
+
+@Injectable({
+    providedIn: 'root',
+})
+export class EloCalculationService {
+
+    public static getNewElos(eloEntry: EloEntry): EloInfoPair {
+        const normalEloDifferences: EloDifferences = EloCalculationService.getNormalEloDifferences(eloEntry);
+        const zeroNewElo: number = EloCalculationService.getActualNewElo(eloEntry.eloInfoPair.playerZero.currentElo,
+                                                                         normalEloDifferences.pointChangeForZero);
+        const oneNewElo: number = EloCalculationService.getActualNewElo(eloEntry.eloInfoPair.playerOne.currentElo,
+                                                                        normalEloDifferences.pointChangeForOne);
+        return {
+            playerZero: {
+                currentElo: zeroNewElo,
+                numberOfGamePlayed: eloEntry.eloInfoPair.playerZero.numberOfGamePlayed + 1,
+            },
+            playerOne: {
+                currentElo: oneNewElo,
+                numberOfGamePlayed: eloEntry.eloInfoPair.playerOne.numberOfGamePlayed + 1,
+            },
+        };
+    }
+    public static getNormalEloDifferences(eloEntry: EloEntry): EloDifferences {
+        const wZero: number = EloCalculationService.getWFrom(eloEntry.winner, Player.ZERO);
+        const wOne: number = EloCalculationService.getWFrom(eloEntry.winner, Player.ONE);
+        const playerZeroInfo: EloInfo = eloEntry.eloInfoPair.playerZero;
+        const playerOneInfo: EloInfo = eloEntry.eloInfoPair.playerOne;
+        const eloZero: number = playerZeroInfo.currentElo;
+        const eloOne: number = eloEntry.eloInfoPair.playerOne.currentElo;
+        const pZero: number = this.getWinningProbability(eloZero, eloOne);
+        const pOne: number = this.getWinningProbability(eloOne, eloZero);
+        const playerZeroK: number = EloCalculationService.getKFrom(playerZeroInfo.numberOfGamePlayed);
+        const playerOneK: number = EloCalculationService.getKFrom(playerOneInfo.numberOfGamePlayed);
+        return {
+            pointChangeForZero: EloCalculationService.getNormalEloDifference(playerZeroK, wZero, pZero),
+            pointChangeForOne: EloCalculationService.getNormalEloDifference(playerOneK, wOne, pOne),
+        };
+    }
+    public static getNormalEloDifference(K: number, W: number, P: number): number {
```

Comment:
what is "normal" ?
Peut-être documenter un poil chacune de ses méthodes avec un commentaire avant disant ce qu'elle fait en une phrase ?

Reply:
petite doc ajoutée, c'est mieux ?

Reply:
je dirais : `(normal as in, without special rules)`, mais j'ai aucune idée c'est quoi ces special rules à ce point !